lib/install/management/create.go

```diff
@@ -211,3 +213,69 @@ func (d *Dispatcher) uploadImages(files map[string]string) error {
 	}
 	return nil
 }
+
+// cleanupAfterCreationFailed cleans up the dangling resource pool for the failed VCH and any bridge network if there is any.
+// The function will not abort and early terminate upon any error during cleanup process. Error details are logged.
+func (d *Dispatcher) cleanupAfterCreationFailed(conf *config.VirtualContainerHostConfigSpec, cleanupNetwork bool) {
+	defer trace.End(trace.Begin("", d.op))
+	var err error
+
+	// clean up dangling empty resource pool
+	d.op.Info("Clean up dangling VCH resource pool after VCH creation failed...")
+
+	if err = d.cleanupEmptyPool(conf); err != nil {
+		d.op.Errorf("Failed to clean up dangling VCH resource pool: %s", err)
+	}
+
+	d.op.Info("Successfully cleaned up dangling resource pool.")
+
+	// only clean up bridge network created if told to
+	if cleanupNetwork {
+		d.op.Info("Cleaning up dangling bridge networks created after VCH creation failed...")
+
+		if err = d.cleanupBridgeNetwork(conf); err != nil {
+			d.op.Errorf("Clean up dangling bridge network returns error: %s", err)
+		}
+
+		d.op.Info("Successfully cleaned up dangling bridge network.")
+	}
+}
+
+// cleanupEmptyPool cleans up any dangling empty VCH resource pool when creating this VCH. no-op when VCH pool is nonempty.
+func (d *Dispatcher) cleanupEmptyPool(conf *config.VirtualContainerHostConfigSpec) error {
+	defer trace.End(trace.Begin("", d.op))
+	var err error
+
+	d.parentResourcepool, err = d.getComputeResource(nil, conf)
+	if err != nil {
+		return err
+	}
+
+	defaultrp, err := d.session.Cluster.ResourcePool(d.op)
+	if err != nil {
+		return err
+	}
+
+	if d.parentResourcepool != nil && d.parentResourcepool.Reference() == defaultrp.Reference() {
+		d.op.Warnf("VCH resource pool is cluster default pool - skipping cleanup")
```

Comment:
Should this actually be a warning? It seems like essentially what we're saying is "Nevermind; there's no cleanup to do."

Reply:
These lines are copied from delete (https://github.com/vmware/vic/blob/master/lib/install/management/delete.go#L114). Not sure what's the reasoning behind printing a warning there, but yes here `d.op.Info` makes more sense.